engine/flink/process/src/main/scala/pl/touk/nussknacker/engine/process/BatchFlinkProcessRegistrar.scala

```diff
@@ -0,0 +1,203 @@
+package pl.touk.nussknacker.engine.process
+
+import java.util.concurrent.TimeUnit
+
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.flink.api.common.functions._
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.api.java.RemoteEnvironment
+import org.apache.flink.api.scala.{ExecutionEnvironment, _}
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.runtime.execution.librarycache.FlinkUserCodeClassLoaders
+import org.apache.flink.streaming.api.windowing.time.Time
+import org.apache.flink.util.Collector
+import pl.touk.nussknacker.engine.Interpreter
+import pl.touk.nussknacker.engine.api._
+import pl.touk.nussknacker.engine.api.context.ValidationContext
+import pl.touk.nussknacker.engine.api.exception.EspExceptionInfo
+import pl.touk.nussknacker.engine.api.test.TestRunId
+import pl.touk.nussknacker.engine.compiledgraph.part._
+import pl.touk.nussknacker.engine.flink.api.process.batch.{FlinkInputFormat, FlinkOutputFormat}
+import pl.touk.nussknacker.engine.flink.util.ContextInitializingFunction
+import pl.touk.nussknacker.engine.flink.util.metrics.InstantRateMeterWithCount
+import pl.touk.nussknacker.engine.graph.EspProcess
+import pl.touk.nussknacker.engine.process.BatchFlinkProcessRegistrar._
+import pl.touk.nussknacker.engine.process.compiler.{CompiledProcessWithDeps, FlinkProcessCompiler}
+import pl.touk.nussknacker.engine.process.util.{MetaDataExtractor, Serializers, UserClassLoader}
+import pl.touk.nussknacker.engine.splittedgraph.end.End
+import pl.touk.nussknacker.engine.splittedgraph.splittednode.SplittedNode
+import pl.touk.nussknacker.engine.util.metrics.RateMeter
+import pl.touk.nussknacker.engine.util.{SynchronousExecutionContext, ThreadUtils}
+
+import scala.concurrent.{Await, ExecutionContext}
+import scala.language.implicitConversions
+import scala.reflect.ClassTag
+import scala.util.control.NonFatal
+
+class BatchFlinkProcessRegistrar(compileProcess: (EspProcess, ProcessVersion) => ClassLoader => CompiledProcessWithDeps,
```

Comment:
Can we at least write somewhere what works and what does not? 
Eg. we have to think about custom transformer API and also about optimizations for SyncInterpretationFunction - I hope we can use mapPartition here

Reply:
I created one big issue #272 and linked this PR there. Also added some TODOs in appropriate places in code.